hazelcast-vnet/azuredeploy-sshPublicKey.json

```diff
@@ -0,0 +1,397 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "newStorageAccountName": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Name for the Storage Account where the Virtual Machine's disks will be placed."
+      }
+    },
+    "location": {
+      "type": "string",
+      "metadata": {
+        "description": "The location where the Virtual Machine will be deployed."
+      },
+      "allowedValues": [
+        "West US",
+        "East US",
+        "Southeast Asia",
+        "East Asia",
+        "West Europe"
+      ]
+    },
+    "adminUserName": {
+      "type": "string",
+      "metadata": {
+        "description": "Username for hazelcast VMs"
+      }
+    },
+    "password": {
+      "type": "string",
+      "metadata": {
+        "description": "SSH Public Key -- not actually a password"
+      }
+    },
+    "clusterUserName": {
+      "type": "string",
+      "metadata": {
+        "description": "Username for hazelcast grid"
+      },
+      "defaultValue": "[parameters('adminUserName')]"
+    },
+    "clusterPassword": {
+      "type": "string",
+      "metadata": {
+        "description": "Password for hazelcast grid"
+      }
+    },
+    "dnsNameForPublicIP": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Name for the Public IP used to access the Virtual Machine."
+      }
+    },
+    "instanceCount": {
+      "type": "int",
+      "metadata": {
+        "description": "The number of hazelcast nodes in the grid"
+      }
+    },
+    "clusterName": {
+      "type": "string",
+      "metadata": {
+        "description": "The tag used to identify this cluster"
+      }
+    },
+    "clusterPort": {
+      "type": "string",
+      "metadata": {
+        "description": "The cluster port to identify this cluster"
+      }
+    },
+    "minHeapSize": {
+      "type": "string",
+      "metadata": {
+        "description": "The minumum heap size for each hazelcast jvm"
+      }
+    },
+    "maxHeapSize": {
+      "type": "string",
+      "metadata": {
+        "description": "The maximum heap size for each hazelcast jvm"
+      }
+    },
+    "aadClientId": {
+      "type": "string",
+      "metadata": {
+        "description": "Azure Active Directory Service Principal client ID"
+      }
+    },
+    "aadClientSecret": {
+      "type": "string",
+      "metadata": {
+        "description": "Azure Active Directory Service Principal client secret"
+      }
+    },
+    "aadTenantId": {
+      "type": "string",
+      "metadata": {
+        "description": "Azure Active Directory Service Principal tenant ID"
+      }
+    },
+    "vmSize": {
+      "type": "string",
+      "metadata": {
+        "description": "The size of each instance VM"
+      },
+      "allowedValues": [
+        "Standard_DS1",
+        "Standard_DS2",
```

Comment:
What's the best guidance here? The most important thing for this scenario is the amount of memory available, but there isn't any reason I'd like to limit the options of the kind of VMs you can use. (Plus, enumerating all possible ones becomes a lengthy list)